test/relations.test.js

```diff
@@ -2569,7 +2569,7 @@ describe('relations', function () {
       Person.create({ name: 'Wilma', addresses: addresses }, function(err, p) {
         err.name.should.equal('ValidationError');
         var expected = 'The `Person` instance is not valid. ';
-        expected += 'Details: `addresses` contains invalid item: `work` (`street` can\'t be blank).';
+        expected += 'Details: `addresses: [ { id: \u001b[32m\'home\'\u001b[39m, street: \u001b[32m\'Home Street\'\u001b[39m },\n  { id: \u001b[32m\'work\'\u001b[39m, street: \u001b[32m\'\'\u001b[39m } ]` contains invalid item: `work` (`street` can\'t be blank).';
```

Comment:
Notice how an invalid item in an array is handled: the message contains `id` of the item, and then information about the error in this item only.

This patch should build on top of that and print only the invalid item in the message. Example:

```
// Option A
Details: `addresses` contains invalid item: `work` (`street: undefined`) can't be blank
```